### PR TITLE
005 competencies per band

### PR DESCRIPTION
**Pull request by Ryan-Jones1999**

### CODE REVIEW

File: src/main/java/com/kainos/ea/service/JobService.java [hunk 1]
```diff
@@ -44,5 +46,22 @@ public List<JobRole> viewJobCapabilities() throws DatabaseException, SQLExceptio
         return jobrole;
     }
 
+    public List<Competency> competency(int bandID) throws DatabaseException, SQLException, NotAValidBandLevelException {
+
+        if(bandID < 0 || bandID > 8){
+            throw new NotAValidBandLevelException();
+        }
+
+        List<Competency> comp = new ArrayList<>();
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/service/JobService.java:55]: Remove redundant variable

File: src/main/java/com/kainos/ea/service/JobService.java [hunk 2]
```diff
@@ -44,5 +46,22 @@ public List<JobRole> viewJobCapabilities() throws DatabaseException, SQLExceptio
...
+
+        comp = jobDao.getCompetency(bandID);
+
+        if(comp.size() <1){
+            throw new DatabaseException(exception);
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/service/JobService.java:60]: Exception without arguments is too broad

File: src/main/java/com/kainos/ea/controller/JobController.java [hunk 3]
```diff
@@ -65,9 +66,20 @@ public Response viewJobCapabilities () {
...
-        return "Hello from a RESTful Web service: " + message;
+    @Path("/viewcompetency/{bandlevelID}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response viewCompentency (@PathParam("bandlevelID") int bandID) {
+        try {
+            return Response.ok(jobService.competency(bandID)).build();
+        } catch (DatabaseException e) {
+            e.printStackTrace();
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+        }catch (SQLException e){
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/JobController.java:77]: Catch branch identical to DatabaseException

File: src/main/java/com/kainos/ea/controller/JobController.java [hunk 4]
```diff
@@ -65,9 +66,20 @@ public Response viewJobCapabilities () {
...
+            e.printStackTrace();
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+        }catch(NotAValidBandLevelException e){
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/controller/JobController.java:80]: catch branch identical to DatabaseException. While here please clean up other APIs doing same thing.

**Review by hellofadude — APPROVED** [review]

**Review by MihneaBocaKainos — APPROVED** [review]

**Pull request merged**